src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -258,7 +258,7 @@ SGVector<T> from_gpu(const SGVector<T>& vec)
 			SG_SERROR("Data memory on GPU but no GPU backend registered. \
 						This can happen if the GPU backend was de-activated \
 						after memory has been transferred to GPU.\n");
-			return NULL;
+			return false;
```

Comment:
boost? we dont have boost :)
`std::optional` is just a c++17 feature, that doesn't mean we cannot have it, question what are the compilers atm that support optional?